Quadratic-Quokkas/FEC-135
# Add Routes.js and remove package lock from git

### Description:

This PR adds a routes.js to serve as the middleman between our client and Atelier's external API. This implements a server route to proxy every request to the atelier api.

<img width="675" alt="Screen Shot 2021-06-06 at 7 16 09 PM" src="https://user-images.githubusercontent.com/49209022/120957368-600d0b00-c70a-11eb-83f7-b9bb61f55a21.png">


Our routes is set up with a `"/*"` - wildcard route. This is capable of capturing the `req.method` (ex: get, post, put) and `req.url` from the incoming requests. Then the server will attach the authentication header with the api keys, turn around and make the same requests to Atelier's external API. Our server will then capture the response data from Atelier and send it back to the client.

`router.all()` - this is used to match and route all the types of http requests from the client
reference: https://expressjs.com/en/guide/routing.html

This PR also removes the `package-lock.json` from our directory - this is because `package-json` had already been committed to the repo before it was added to `.gitignore`, so `gitignore` for this file does not work unless we delete it first.

### QA Steps:
1. Add my forked repo as remote
2. `git fetch irisj23 main` to fetch my updates
3. check into my branch `git checkout irisj23/main`
4.  might need to re-run webpack, make sure your server is running
5. test the routes below - Get routes can be tested in browser or postman, Post routes will need to be tested with postman only

### Notes:

I have tested the following:

**PRODUCTS:**

Get:
Get list of all products - `http://localhost:3000/api/products`
Get one product - `http://localhost:3000/api/products/:product_id`
Get all styles for one product - `http://localhost:3000/api/products/:product_id/styles`
Get id's of related products - `http://localhost:3000/api/products/:product_id/related`


**REVIEWS:**

Get:
Get list of reviews for a given product - `http://localhost:3000/api/reviews?product_id=25171`
Get review metadata for given product - `http://localhost:3000/api/reviews/meta?product_id=25171`

Post (tested with postman):
Add a review for given product (needs body parameters) - `localhost:3000/api/reviews`

Put (tested with postman):
Updates a review to show it was helpful - `localhost:3000/api/reviews/348648/helpful`
Update a review to show it was reported - `localhost:3000/api/reviews/348648/report`

**QUESTIONS / ANSWERS:**

Get:
Get a list of questions for a given product - `http://localhost:3000/api/qa/questions?product_id=25171`
Get a list of answers for a given question - `http://localhost:3000/api/qa/questions/183526/answers`

Post (tested with postman):
Adds a question for given product - `localhost:3000/api/qa/questions`
Adds a answer for given question - `localhost:3000/api/qa/questions/183526/answers`

Put (tested with postman)
Updates a question to mark as helpful - `localhost:3000/api/qa/questions/183526/helpful`
Updates a answer to mark as helpful - `localhost:3000/api/qa/answers/1991254/helpful`
*did not test put for reporting a question and answer 


**CART:**

Get:
Gets a list of products added to cart by user - `http://localhost:3000/api/cart`

Post (test with postman):
Adds a product to cart - `localhost:3000/api/cart`
**needs specific sku id**

**INTERACTIONS:**

Post (tested with postman):
Adds a interaction to db - `localhost:3000/api/interactions`
